### PR TITLE
feat: add support for clang 10 (which requires ubuntu 18)

### PR DESCRIPTION
**Pull request by joshessman-llnl**
LLVM's only Ubuntu release is for 18.04, so I created a new 18.04 base image with the latest version of CMake (3.18), which has C++17/20 support.

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
Looks good -  do you think we should update the base for the other images?

**Comment by joshessman-llnl** [comment]
clang-9 is the only other version of clang with an Ubuntu 18 binary release, so possibly that one could be ported, but I think there's value in maintaining the older images.  NVCC 11 and GCC 9 should definitely use 18.04, when those become necessary.

**Comment by white238** [comment]
I would leave the older images alone and only use the newer compilers on the new base.  For backwards compatibility (for a while) as well as just not creating more work for yourself for compilers that will be going away soonish.

File: ubuntu-18-base/generic-setup.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+set -e
+set -x
+
+apt-get -qq update
+apt-get -qq install -y --no-install-recommends wget python3-dev python3-pip build-essential sudo git vim dh-autoreconf ninja-build ca-certificates valgrind
```
Comment (white238) [inline ubuntu-18-base/generic-setup.sh:6]: I wonder if we should start adding some more useful packages to some of these docker images, not sure if the base one is the best place to put them.  But packages like openmpi..

**Review by white238 — APPROVED** [review]

File: ubuntu-18-base/generic-setup.sh (hunk 1 above)
Comment (cyrush) [reply ubuntu-18-base/generic-setup.sh:6]: https://github.com/Alpine-DAV/ascent/blob/4dcab4aa30c0e4fd23b6fa03ef5f00d3db16833a/src/examples/docker/ubuntu/Dockerfile#L49

Comment (cyrush) [reply ubuntu-18-base/generic-setup.sh:6]: includes many common deps in case it helps

**Review by rhornung67 — APPROVED** [review]

File: clang-10/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM axom/compilers:ubuntu-18-base
+
+LABEL maintainer="David Beckingsale <david@llnl.gov>"
```
Comment (davidbeckingsale) [inline clang-10/Dockerfile]: Feel free to change the maintainer to you 😄

**Pull request merged**